requests/adapters.py

```diff
@@ -388,7 +388,10 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                     low_conn.endheaders()
 
                     for i in request.body:
-                        low_conn.send(hex(len(i))[2:].encode('utf-8'))
+                        chunk_size = len(i)
+                        if chunk_size == 0:
+                            break
```

Comment:
break - making it consistent with the existing behavior where the connection would have anyway terminated upon sending this zero size chunk.